service/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateLandingZoneResourcesFlightStep.java

```diff
@@ -0,0 +1,98 @@
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class CreateLandingZoneResourcesFlightStep implements Step {
+
+  private final LandingZoneService landingZoneService;
+  private final LandingZoneRequest landingZoneRequest;
+  private final String jobIdKey;
+
+  public CreateLandingZoneResourcesFlightStep(
+      LandingZoneService landingZoneService,
+      LandingZoneRequest landingZoneRequest,
+      String jobIdKey) {
+    this.landingZoneService = landingZoneService;
+    this.landingZoneRequest = landingZoneRequest;
+    this.jobIdKey = jobIdKey;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    FlightUtils.validateRequiredEntries(
+        context.getInputParameters(),
+        LandingZoneFlightMapKeys.LANDING_ZONE_ID,
+        LandingZoneFlightMapKeys.BEARER_TOKEN,
+        JobMapKeys.RESULT_PATH.getKeyName());
+
+    FlightUtils.validateRequiredEntries(
+        context.getWorkingMap(), LandingZoneFlightMapKeys.BILLING_PROFILE);
+
+    var landingZoneId =
+        context.getInputParameters().get(LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+
+    var subflightId = getSubflightId(landingZoneId.toString());
+    if (isFlightAlreadyExists(context, subflightId)) {
+      return StepResult.getStepResultSuccess();
+    }
+
+    // use token from parent flight
+    var bearerToken =
+        context.getInputParameters().get(LandingZoneFlightMapKeys.BEARER_TOKEN, BearerToken.class);
+    var resultPath =
+        context.getInputParameters().get(JobMapKeys.RESULT_PATH.getKeyName(), String.class);
+    var billingProfile =
+        context.getWorkingMap().get(LandingZoneFlightMapKeys.BILLING_PROFILE, ProfileModel.class);
+
+    // this parameter should be read in AwayCreatLandingZoneFlightStep
+    context.getWorkingMap().put(jobIdKey, subflightId);
+
+    // create sub-flight, which is supposed to create all required Azure resources
+    landingZoneService.startLandingZoneResourceCreationJob(
+        subflightId,
+        landingZoneRequest,
+        billingProfile,
+        landingZoneId,
+        bearerToken,
+        resultPath + subflightId);
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    // nothing to undo
+    return StepResult.getStepResultSuccess();
+  }
+
+  private boolean isFlightAlreadyExists(FlightContext context, String flightId)
+      throws InterruptedException {
+    boolean flightAlreadyExists = true;
+    try {
+      context.getStairway().getFlightState(flightId);
+    } catch (FlightNotFoundException e) {
+      flightAlreadyExists = false;
+    }
+    return flightAlreadyExists;
+  }
+
+  private String getSubflightId(String landingZoneId) {
+    // subFlightId is limited up to 36 characters
+    // use first 8 characters and last 4 characters of landingzoneId
```

Comment:
So there is the possibility of non-unique IDs. What happens in that case? And is it worth decreasing the odds by increasing the number of characters used?

Reply:
If such thing happened a flight failed to start. I am using 12 characters for id. 10 digits and 26 characters. The name collision is possible, but the probability is really low (something like 36 to the power of -12)